frontend/src/components/widgets/CameraInput/WebcamComponent.tsx

```diff
@@ -0,0 +1,174 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Video } from "@emotion-icons/open-iconic"
+import { SwitchCamera } from "@emotion-icons/material-rounded"
+import React, { ReactElement, useState, useRef } from "react"
+import { isMobile } from "react-device-detect"
+import Webcam from "react-webcam"
+import { useTheme } from "emotion-theming"
+import { Theme } from "src/theme"
+
+import Button, { Kind } from "src/components/shared/Button"
+import Tooltip, { Placement } from "src/components/shared/Tooltip"
+import Icon from "src/components/shared/Icon"
+import themeColors from "src/theme/baseTheme/themeColors"
+import CameraInputButton from "./CameraInputButton"
+import {
+  StyledBox,
+  StyledCameraInput,
+  StyledDescription,
+  StyledLink,
+  StyledSwitchFacingModeButton,
+} from "./styled-components"
+
+export interface Props {
+  handleCapture: (capturedPhoto: string | null) => void
+  width: number
+  disabled: boolean
+  inClearPhotoMode: boolean
+  setInClearPhotoMode: (inClearPhotoMode: boolean) => void
+}
+
+enum FacingMode {
+  USER = "user",
+  ENVIRONMENT = "environment",
+}
+
+enum WebcamPermission {
+  PENDING = "pending",
+  SUCCESS = "success",
+  ERROR = "error",
+}
+
+const WebcamComponent = ({
+  handleCapture,
+  width,
+  disabled,
+  inClearPhotoMode,
+  setInClearPhotoMode,
+}: Props): ReactElement => {
+  const [webcamPermission, setWebcamRequestState] = useState(
+    WebcamPermission.PENDING
+  )
+  const videoRef = useRef<Webcam>(null)
+  const [facingMode, setFacingMode] = useState(FacingMode.USER)
+
+  function capture(): void {
+    if (videoRef.current !== null) {
+      const imageSrc = videoRef.current.getScreenshot()
+      handleCapture(imageSrc)
+    }
+  }
+
+  function switchCamera(): void {
+    setFacingMode(prevState =>
+      prevState === FacingMode.USER ? FacingMode.ENVIRONMENT : FacingMode.USER
+    )
+  }
+
+  const theme: Theme = useTheme()
+
+  return (
+    <StyledCameraInput width={width}>
+      {webcamPermission !== WebcamPermission.SUCCESS &&
+      !disabled &&
+      !inClearPhotoMode ? (
+        <AskForCameraPermission width={width} />
+      ) : (
+        isMobile && (
+          <StyledSwitchFacingModeButton>
+            <Tooltip content={"Switch camera"} placement={Placement.TOP_RIGHT}>
+              <Button kind={Kind.MINIMAL} onClick={switchCamera}>
+                <Icon
+                  content={SwitchCamera}
+                  size="twoXL"
+                  color={themeColors.white}
+                />
+              </Button>
+            </Tooltip>
+          </StyledSwitchFacingModeButton>
+        )
+      )}
+      <StyledBox
+        hidden={
+          webcamPermission !== WebcamPermission.SUCCESS &&
+          !disabled &&
+          !inClearPhotoMode
```

Comment:
nit: I believe this is the same value as the boolean computed in the ternary expression above, so this can be factored out along with it